Xamarin.Forms.Platform.Android/AppCompat/FormsAppCompatActivity.cs

```diff
@@ -180,19 +182,61 @@ protected override void OnCreate(Bundle savedInstanceState)
 			OnStateChanged();
 
 			AddStatusBarUnderlay();
+			Window.DecorView.ViewTreeObserver.GlobalLayout += ViewTreeObserverOnGlobalLayout;
 		}
 
 		protected override void OnDestroy()
 		{
 			// may never be called
 			base.OnDestroy();
 
+			Window.DecorView.ViewTreeObserver.GlobalLayout -= ViewTreeObserverOnGlobalLayout;
+
 			MessagingCenter.Unsubscribe<Page, AlertArguments>(this, Page.AlertSignalName);
 			MessagingCenter.Unsubscribe<Page, bool>(this, Page.BusySetSignalName);
 			MessagingCenter.Unsubscribe<Page, ActionSheetArguments>(this, Page.ActionSheetSignalName);
 
-			if (_platform != null)
-				_platform.Dispose();
+			_platform?.Dispose();
+		}
+
+		void ViewTreeObserverOnGlobalLayout(object sender, EventArgs eventArgs)
```

Comment:
This handler receives global layout changes. Android has this weird design rule that fullscreen does not resize. Most developers get around this issue by adding or removing padding on the content based on visible display area.